include/El/core/AxpyInterface.hpp

```diff
@@ -41,48 +41,66 @@ class AxpyInterface
     void Detach();
 
 private:
+#if MPI_VERSION>=3 && defined(EL_USE_IBARRIER_FOR_AXPY)
+    static const Int 
+        DATA_TAG        =1, 
+        DATA_REQUEST_TAG=2, 
+        DATA_REPLY_TAG  =3;
+#else
     static const Int 
         DATA_TAG        =1, 
         EOM_TAG         =2, 
         DATA_REQUEST_TAG=3, 
         DATA_REPLY_TAG  =4;
+#endif
 
+    //request object for polling on Issends
     bool attachedForLocalToGlobal_, attachedForGlobalToLocal_;
+
     DistMatrix<T,MC,MR>* localToGlobalMat_;
     const DistMatrix<T,MC,MR>* globalToLocalMat_;
 
-    vector<bool> sentEomTo_, haveEomFrom_;
-    vector<byte> recvVector_;
-    vector<mpi::Request> eomSendRequests_;
-
-    vector<deque<vector<byte>>> dataVectors_, requestVectors_, replyVectors_;
-    vector<deque<bool>> sendingData_, sendingRequest_, sendingReply_;
-    vector<deque<mpi::Request>> 
+#if MPI_VERSION>=3 && defined(EL_USE_IBARRIER_FOR_AXPY)
+#else
+    std::vector<bool> sentEomTo_, haveEomFrom_;
+    std::vector<mpi::Request> eomSendRequests_;
+#endif    
+    
+    std::vector<std::deque<bool>> 
+        sendingData_, sendingRequest_, sendingReply_;
+    std::vector<std::deque<mpi::Request>> 
```

Comment:
std::vector and std::deque were imported into the El namespace, and so it is now okay (and preferred) to use just "vector" and "deque" (the same holds for "cout", "cerr", and "endl").